Implement groupUniqArrayArrayMap as SimpleAggregateFunction

mdonkers opened the pull request:
Implementation for https://github.com/ClickHouse/ClickHouse/issues/74847

### Changelog category (leave one):
- Improvement


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for `groupUniqArrayArrayMap` as `SimpleAggregateFunction`

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)


#### CI Settings (Only check the boxes if you know what you are doing)

All builds in Builds_1 and Builds_2 stages are always mandatory and will run independently of the checks below:
- [x] <!---ci_include_stateless--> Only: Stateless tests
- [ ] <!---ci_include_integration--> Only: Integration tests
- [ ] <!---ci_include_performance--> Only: Performance tests
---
- [ ] <!---ci_exclude_style--> Skip: Style check
- [ ] <!---ci_exclude_fast--> Skip: Fast test
---
- [ ] <!---woolen_wolfdog--> Run all checks ignoring all possible failures (Resource-intensive. All test jobs execute in parallel).
- [ ] <!---no_ci_cache--> Disable CI cache

mdonkers committed:
Implement groupUniqArrayArrayMap as SimpleAggregateFunction

robot-ch-test-poll commented:
<!-- automatic status comment for PR #75034 from mdonkers/ClickHouse:master -->
*This is an automated comment for commit 585683ec1f9db2b53c0bdb16bb50c9ca510dcf45 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/75034/585683ec1f9db2b53c0bdb16bb50c9ca510dcf45/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/75034/585683ec1f9db2b53c0bdb16bb50c9ca510dcf45/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

mdonkers commented:
@nikitamikhaylov I've reverted the `tests/queries/0_stateless/01570_aggregator_combinator_simple_state.sql` test, because it was failing with the message
```
Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Aggregation 'groupUniqArrayArrayMap' is not implemented for mapped arrays. (ILLEGAL_TYPE_OF_ARGUMENT)
```
and I'm wondering if the way I added it would work at all, or if it needs some other syntax:
```
with groupUniqArrayArrayMapSimpleState(([number], [[number]])) as c select toTypeName(c), c from numbers(1);
```

Happy to add it back if needed, and if you know what the syntax should be :-)

mdonkers commented:
Also not sure why a single test is still failing. Doesn't seem to be related to these changes?

alexkats reviewed (commented):
LGTM, a couple of nitpicks

alexkats commented:
> @nikitamikhaylov I've reverted the `tests/queries/0_stateless/01570_aggregator_combinator_simple_state.sql` test, because it was failing with the message
> 
> ```
> Code: 43. DB::Exception: Received from localhost:9000. DB::Exception: Aggregation 'groupUniqArrayArrayMap' is not implemented for mapped arrays. (ILLEGAL_TYPE_OF_ARGUMENT)
> ```
> 
> and I'm wondering if the way I added it would work at all, or if it needs some other syntax:
> 
> ```
> with groupUniqArrayArrayMapSimpleState(([number], [[number]])) as c select toTypeName(c), c from numbers(1);
> ```
> 
> Happy to add it back if needed, and if you know what the syntax should be :-)

From what I see, it won't work, so don't worry about this test

alexkats commented:
> Also not sure why a single test is still failing. Doesn't seem to be related to these changes?

Took a look, and the failure is not related

mdonkers commented:
Thanks for the review @alexkats 

I've fixed the formatting issues. But also squashed the commits to have a clean history.

alexkats commented:
Stateless tests (aarch64) - `02782_uniq_exact_parallel_merging_bug` - fixed in https://github.com/ClickHouse/ClickHouse/pull/75222
